frontend/server/src/DAO/Problems.php

```diff
@@ -163,23 +163,22 @@ class Problems extends \OmegaUp\DAO\Base\Problems {
         }
 
         $difficulties = ['easy', 'medium', 'hard'];
-        $difficultyConditions = 'p.difficulty >= ? AND p.difficulty <= ?';
 
         if (in_array($difficulty, $difficulties, true)) {
             if ($difficulty === $difficulties[0]) {
                 $clauses[] = [
-                    $difficultyConditions,
-                    ['0', '1.33'],
+                    'p.difficulty < ?',
+                    ['1.34'],
```

Comment:
```suggestion
                    'p.difficulty < 1.34',
                    [],
```

aquí no hay necesidad de hacerle bind a las "variables" porque son constantes. eso también evita que se pasen como cadenas y se conviertan a números flotantes.

igual abajo.

Reply:
Salía algo de `string is always defined and non-null`. Intenté haciendo que siempre fuera `string` y no `null|string` pero creo que no era eso.

Reply:
puedes copiar y pegar el error exacto en vez de parafrasearlo? hay muchos detalles que faltan :S

Reply:
`ERROR: TypeDoesNotContainType - frontend/server/src/Controllers/Problem.php:6048:13 - string is always defined and non-null (see https://psalm.dev/056)
            $difficulty ?? 'all'`

Reply:
AH! en esos casos no se necesita el `??` ni el `?:`. puedes cambiar todos los `?:` de este cambio a `??` y en los que te salga ese error en particular, quitar el `??`?